url_scraper.py

```diff
@@ -0,0 +1,60 @@
+import requests
+import time
+import random
+from bs4 import BeautifulSoup
+
+
+base_url = "https://www.foodnetwork.com/recipes/recipes-a-z/"
+
+# All categories of alphabetical recipes from the base url
+url_extensions = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
+                  'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'xyz', '123']
+
+
+# Finds length of a given page, to ensure that all recipes are scraped for each category
+def page_search_length(url):
+    r = requests.get(url)
+    soup = BeautifulSoup(r.text, 'lxml')
+
+    lengths = soup.findAll('li', {"class": "o-Pagination__a-ListItem"})
+    
+    # If had to stop in process of scraping, resume from most recently printed extension
+    print(url)
+    return int(lengths[-2].find('a').get_text())
+
+# Gets all recipe urls from each page of each category, and then writes them into the text file
+def subpage_search(url, pages):
+
+    # Write urls to text file for later
+    text_urls = open('recipe_urls.txt', 'a')
+    url_list = []
+
+    for x in range(1, pages + 1):
+        search_url = url + str(x)
+        r = requests.get(search_url)
```

Comment:
when the code is ran there's no checking to make sure that the page exists (this can be done with getting the request object and set it to .status_code, which will return 200 if it's a real working page, and 404 if it doesn't exist ) so while the urls are already set it could be good if we have to run it again later